File: 35. Search Insert Position/python/attempt_1_test.py
from attempt_1 import Solution

def test_attempt_1():
    # test 1
    nums = [1,3,5,6]
    target = 5

    res = Solution.searchInsert(self=Solution, nums=nums, target=target)

    assert res == 2
    
    # test 2
    nums = [1,3,5,6]
    target = 2

    res = Solution.searchInsert(self=Solution, nums=nums, target=target)

    assert res == 1
    
    # test 3
    nums = [1,3,5,6]
    target = 7

    res = Solution.searchInsert(self=Solution, nums=nums, target=target)

    assert res == 4

    # test 4
    nums = [1]
    target = 0
    
    res = Solution.searchInsert(self=Solution, nums=nums, target=target)

    assert res == 0
    
    # test 5
    nums = [1,3]
    target = 3
    
    res = Solution.searchInsert(self=Solution, nums=nums, target=target)
    
    assert res == 1
    
    # test 6
    nums = [1]
    target = 1
    res = Solution.searchInsert(self=Solution, nums=nums, target=target)

    assert res == 0

File: 209. Minimum Size Subarray Sum/typescript/attempt_1.test.ts
import minSubArrayLen from './attempt_1';

describe("test minSubArrayLen", () => {
  test("Test 1 - target=7, nums=[2,3,1,2,4,3]", () => {
    expect(minSubArrayLen(7, [2, 3, 1, 2, 4, 3])).toBe(2);
  });

  test("Test 2 - target=4, nums=[1,4,4]", () => {
    expect(minSubArrayLen(4, [1, 4, 4])).toBe(1);
  });

  test("Test 3 - target=11, nums=[1,1,1,1,1,1,1,1]", () => {
    expect(minSubArrayLen(11, [1, 1, 1, 1, 1, 1, 1, 1])).toBe(0);
  });
});
